### ISSUE TITLE
[java] LawOfDemeter: False positive when casting to derived class

### ISSUE THREAD

**Issue by Fernal73** [pmd/pmd#2189]
https://github.com/pmd/pmd/issues/2160#issuecomment-565639733

**Affects PMD Version:** 
6.20.0
**Rule:**
LawOfDemeter
**Description:**
Casting to derived class triggers method chaining error for this rule.
**Code Sample demonstrating the issue:**

```java
package pmdtests;


  class TaskManager {
   
    private final List<ForkJoinTask<Integer>> tasks;

  
    public void cancelTasks(ForkJoinTask<Integer> cancelTask) {
      for (ForkJoinTask<Integer> task : tasks) {
        if (!task.equals(cancelTask)) {
          task.cancel(true);
          ((SearchNumberTask) task).writeCancelMessage();
        }
      }
    }
}
 

```

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*
